esyi/Resolver.ml

```diff
@@ -129,7 +131,22 @@ let setOCamlVersion ocamlVersion resolver =
   resolver.ocamlVersion <- Some ocamlVersion
 
 let setResolutions resolutions resolver =
-  resolver.resolutions <- resolutions
+  (* First we set the resolutions *)
+  resolver.resolutions <- resolutions;
+  (* Then we set the usage of every resolution to 0 *)
+  List.iter
+    ~f:(fun r -> Hashtbl.add resolver.resolutionUsage r false)
+    (Resolutions.entries resolutions)
```

Comment:
Great!

One small tweak we can make to simplify the code — instead of populating hashtable with `false` values we can assume that if key isn't in the table — resolution is unused. We can just init `resolutionUsage` to be empty here instead.

Then `getUnusedResolutions` should iterate over resolutions and use `Hastbl.find_opt` to check for `Some _` or `None`.